server/src/main/java/org/candlepin/resource/ConsumerResource.java

```diff
@@ -1397,6 +1377,63 @@ public boolean performConsumerUpdates(ConsumerDTO updated, Consumer toUpdate,
         return changesMade;
     }
 
+    private boolean updateEnvironment(ConsumerDTO updated, Consumer toUpdate) {
+        boolean changesMade = false;
+        boolean environmentFound = false;
+        Environment e = null;
+        String environmentId = updated.getEnvironment() == null ? null : updated.getEnvironment().getId();
+        String environmentName = updated.getEnvironment() == null ? null : updated.getEnvironment().getName();
+
+        if (environmentName != null && environmentId == null) {
+            List<Environment> environment = environmentCurator.listForOwnerByName(toUpdate.getOwner(),
+                environmentName).list();
+
+            if (!environment.isEmpty() && environment.size() == 1) {
+                e = environment.get(0);
+                environmentId = environment.get(0).getId();
+                environmentFound = true;
+            }
+            else {
+                throw new NotFoundException(i18n.tr(
+                        "Environment with name \"{0}\" could not be found.", environmentName));
+            }
+
+        }
+
+        if (environmentId != null && (toUpdate.getEnvironmentId() == null ||
+            !toUpdate.getEnvironmentId().equals(environmentId))) {
+
+            if (!environmentFound) {
+                e = environmentCurator.get(environmentId);
+
+                if (e == null) {
+                    throw new NotFoundException(i18n.tr(
+                            "Environment with ID \"{0}\" could not be found.", environmentId));
+                }
+            }
+
+            changesMade = true;
+
+            log.info("Updating environment to: {}", environmentId);
+        }
+
+        if (e != null && changesMade) {
+            toUpdate.setEnvironment(e);
+
+            // reset content access cert
+            Owner owner = ownerCurator.findOwnerById(toUpdate.getOwnerId());
+            if (owner.isContentAccessEnabled()) {
+                toUpdate.setContentAccessCert(null);
+                contentAccessCertService.removeContentAccessCert(toUpdate);
+            }
+            // lazily regenerate certs, so the client can still work
+            poolManager.regenerateCertificatesOf(toUpdate, true);
+
```

Comment:
Move this whitespace up two lines to logically separate the regeneration section and the CAC section.

Reply:
corrected.